classes/security/authorization/internal/ApiAuthorizationMiddleware.inc.php

```diff
@@ -38,6 +38,14 @@ protected function _authorize($slimRequest) {
 		$this->_handler->setSlimRequest($slimRequest);
 		$request = $this->_handler->getRequest();
 		$args = array($slimRequest);
+
+		if ($slimRequest->getQueryParam('apiToken', null) === null) {
+			return $request->getRouter()->handleAuthorizationFailure(
+				$request,
+				'api.401.forgottenApiToken'
+			);
+		}
```

Comment:
This is going to be the source of your test failures. If you try to load the submissions page, you'll see the `api.401.forgottenApiToken` error. That's because it's possible to use the API without an API token when using the PHP session cookies (ie - requests from the same domain). This is how the editorial backend works.

I know I told you to add this, but in hindsight let's remove it. I don't want to get too deep in the weeds inspecting where the request came from at this point.

Reply:
Ok! But what do you think about keeping this in our radar? To refactor it in the following months.

Reply:
Sure!